python/tank/platform/engine.py

```diff
@@ -1832,14 +1836,69 @@ def start_engine(engine_name, tk, context):
     Raises TankEngineInitError if an engine could not be started
     for the passed context.
     """
+    return _start_engine(engine_name, tk, None, context)
+
+
+def _restart_engine(new_context):
+
+    engine = current_engine()
+    try:
+        # Track some of the current state before restarting the engine.
+        old_context = engine.context
+
+        # Restart the engine. If we were given a new context to use,
+        # use it, otherwise restart using the same context as before.
+        new_context = new_context or engine.context
```

Comment:
I don't like the init term personally. It implies it's something being done once and then doesn't change, which is not the case here.

An engine is bound to a specific context, which is why I think I prefer EngineContextBindHook.

The fact that the term bind is neutral compared to change means that it doesn't preclude binding the same value twice. Maybe we can propose the term during Dailies today?

If I understand correctly your comments, this is therefore not a change in design and only in terminology, right?